`datetime.astimezone()` incorrectly requires a `tz` argument

The `datetime.astimezone` method, according to DateType, requires a `tz` argument. This doesn't match the behaviour in the `datetime` module: per https://docs.python.org/3/library/datetime.html#datetime.datetime.astimezone as of Python v3.3, omitting `tz` or specifying `tz=None` will obtain the timezone to use from the OS.

Test case below; everything works as expected at runtime, but mypy complains about the lack of `tz` argument when type checking:

```python3
from typing import reveal_type

import datetime

import datetype

print('Using datetype:')
n = datetype.NaiveDateTime.now()
print(n)
reveal_type(n)
a = n.astimezone()
print(a)
reveal_type(a)

print()
print('Using datetime:')
nn = datetime.datetime.now()
print(nn)
reveal_type(nn)
aa = nn.astimezone()
print(aa)
reveal_type(aa)
```

```
$ python3 test.py
Using datetype:
2023-09-24 23:12:09.278491
Runtime type is 'datetime'
2023-09-24 23:12:09.278491+01:00
Runtime type is 'datetime'

Using datetime:
2023-09-24 23:12:09.278858
Runtime type is 'datetime'
2023-09-24 23:12:09.278858+01:00
Runtime type is 'datetime'
```

```
 $ mypy --strict test.py
test.py:10: note: Revealed type is "datetype.DateTime[None]"
test.py:11: error: Missing positional argument "tz" in call to "astimezone" of "DateTime"  [call-arg]
test.py:11: error: Need type annotation for "a"  [var-annotated]
test.py:13: note: Revealed type is "datetype.DateTime[Any]"
test.py:19: note: Revealed type is "datetime.datetime"
test.py:22: note: Revealed type is "datetime.datetime"
Found 2 errors in 1 file (checked 1 source file)
```

--- a/expected_mypy.txt
+++ b/expected_mypy.txt
@@ -36,4 +36,11 @@ tryit.py:46: note:     Got:
 tryit.py:46: note:         def timetz(self) -> DateTime[None]
 tryit.py:46: note:     tzinfo: expected "tzinfo", got "None"
 tryit.py:48: error: "tzinfo" has no attribute "key"  [attr-defined]
-Found 8 errors in 1 file (checked 1 source file)
+tryit.py:53: error: Incompatible types in assignment (expression has type "DateTime[timezone]", variable has type "NaiveDateTime")  [assignment]
+tryit.py:53: note: Following member(s) of "DateTime[timezone]" have conflicts:
+tryit.py:53: note:     Expected:
+tryit.py:53: note:         def timetz(self) -> DateTime[None]
+tryit.py:53: note:     Got:
+tryit.py:53: note:         def timetz(self) -> DateTime[timezone]
+tryit.py:53: note:     tzinfo: expected "None", got "timezone"
+Found 9 errors in 1 file (checked 1 source file)

--- a/src/datetype/__init__.py
+++ b/src/datetype/__init__.py
@@ -7,6 +7,7 @@ from datetime import (
     time as _time,
     timedelta as _timedelta,
     tzinfo as _tzinfo,
+    timezone as _timezone,
 )
 from time import struct_time
 from typing import (
@@ -475,6 +476,11 @@ class DateTime(Protocol[_GMaybeTZDT]):
     ) -> Self:
         "If no replacement tz is specified then we inherit"
 
+    @overload
+    def astimezone(self, tz: None = None) -> DateTime[_timezone]:
+        ...
+
+    @overload
     def astimezone(self, tz: _FuncTZ) -> DateTime[_FuncTZ]:
         ...
 

--- a/tryit.py
+++ b/tryit.py
@@ -49,3 +49,6 @@ y.tzinfo.key                    # error; no attribute defined
 
 specific: DateTime[ZoneInfo] = DateTime.now(ZoneInfo("America/Los_Angeles"))
 specific.tzinfo.key             # ok; ZoneInfo has 'key' attribute
+
+not_naive: NaiveDateTime = a.astimezone()
+is_aware: DateTime[timezone] = a.astimezone(None)

